src/main/app/src/app/shared/utils/form-data.ts

```diff
@@ -0,0 +1,46 @@
+type FormDataMappingFunction<T> = (
+  keyValuePair: readonly [string, T],
+) => readonly [string, string | Blob] | readonly [string, Blob, string];
+
+type FormDataMapper<T> = Partial<{
+  [K in keyof T]: T[K] extends string | Blob | { toString: () => string }
+    ? FormDataMappingFunction<T[K]> | true
+    : FormDataMappingFunction<T[K]>;
+}>;
+
+function parseFormValue(v: unknown) {
+  if (!v) return false;
+  if (typeof v === "string" || v instanceof Blob) return v;
+  if (v.toString && typeof v.toString === "function") return v.toString();
+  return false;
```

Comment:
Awesome tests, just missing this line... can't we get here? Or can you create a test that documents a case where the value can't be parsed?